index.html

```diff
@@ -102,29 +109,79 @@ <h2>Items in cart</h2>
       function initializeCart() {
         const itemsInCart = []
 
+        const cartListElement = document.getElementById('cart-list')
+
+        function addToCart(array, item) {
+          array.push({
+          product: item.product,
+          price: item.price,
+          count: 1,
+          })
+        }
+
         Array.from(document.querySelectorAll('.add-to-cart')).forEach(
           (button) => {
             button.addEventListener('click', () => {
               console.log('clicked add to cart')
+              
+              const typesOfItemsInCart = itemsInCart.length
 
-              itemsInCart.push({
+              const latestCartItem = {
                 product: button.dataset.product,
                 price: Number(button.dataset.price),
-              })
-
-              const itemsInCartCount = itemsInCart.length
+              }
+
+              // This will keep track if the prodcut has been added to itemsInCart
+              let isInCart = false
+
+              // Check if the cart is empty
+              if (itemsInCart.length === 0) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              } else {
+                // Cart isn't empty, so is the same type of product already exist?
+                for (let i=0; i < typesOfItemsInCart; i++) {
+                  if (itemsInCart[i].product === latestCartItem.product) {
+                    itemsInCart[i].count++
+                    isInCart = true
+                    break
+                  }
+                }
+              }
```

Comment:
This is great! You can simplify things further by keeping just the lines inside the `else` block (lines 142-149). This still works because:
1. the `for` loop will exit immediately when the array is empty (since there's no items to iterate over), which means you can get rid of the check for whether the cart is empty on line 138
2. if the cart is empty, your check for `isInCart` on line 153 will still be relevant, so the first product will still get added.
```suggestion
              for (let i = 0; i < typesOfItemsInCart; i++) {
                if (itemsInCart[i].product === latestCartItem.product) {
                  itemsInCart[i].count++
                  isInCart = true
                  break
                }
              }
```